src/entitysdk/client.py

```diff
@@ -248,4 +379,107 @@ def update_entity(
 
     json_data = response.json()
 
-    return deserialize_entity(json_data, entity_type)
+    return serdes.deserialize_entity(json_data, entity_type)
+
+
+def upload_asset_file(
+    url: str,
+    *,
+    asset_path: Path,
+    asset_metadata: LocalAssetMetadata,
+    project_context: ProjectContext,
+    token: str,
+    http_client: httpx.Client | None = None,
+):
+    """Upload asset to an existing entity's endpoint from a file path."""
+    with open(asset_path, "rb") as file_content:
+        return upload_asset_content(
+            url=url,
+            asset_content=file_content,
+            asset_metadata=asset_metadata,
+            project_context=project_context,
+            token=token,
+            http_client=http_client,
+        )
+
+
+def upload_asset_content(
+    url: str,
+    *,
+    asset_content: io.BufferedIOBase,
+    asset_metadata: LocalAssetMetadata,
+    project_context: ProjectContext,
+    token: str,
+    http_client: httpx.Client | None = None,
+) -> Asset:
+    """Upload asset to an existing entity's endpoint from a file-like object."""
+    files = {
+        "file": (
+            asset_metadata.file_name,
+            asset_content,
+            asset_metadata.content_type,
+        )
+    }
+    response = make_db_api_request(
+        url=url,
+        method="POST",
+        files=files,
+        project_context=project_context,
+        token=token,
+        http_client=http_client,
+    )
+    return serdes.deserialize_entity(response.json(), Asset)
+
+
+def download_asset_content(
+    url: str,
+    *,
+    project_context: ProjectContext,
+    token: str,
+    http_client: httpx.Client | None = None,
+) -> bytes:
+    """Download asset content.
+
+    Args:
+        url: URL of the asset.
+        project_context: Project context.
+        token: Authorization access token.
+        http_client: HTTP client.
+
+    Returns:
+        Asset content in bytes.
+    """
+    response = make_db_api_request(
+        url=url,
+        method="GET",
+        project_context=project_context,
+        token=token,
+        http_client=http_client,
+    )
+    return response.content
+
+
+def download_asset_file(
+    url: str,
+    *,
+    output_path: Path,
+    project_context: ProjectContext,
+    token: str,
+    http_client: httpx.Client | None = None,
+) -> None:
+    """Download asset file to a file path.
+
+    Args:
+        url: URL of the asset.
+        output_path: Path to save the file to.
+        project_context: Project context.
+        token: Authorization access token.
+        http_client: HTTP client.
+    """
+    bytes_content = download_asset_content(
+        url=url,
+        project_context=project_context,
+        token=token,
+        http_client=http_client,
+    )
+    output_path.write_bytes(bytes_content)
```

Comment:
Picky, the order of the download functions can be swapped to match the order of the upload functions, as in:
upload_asset_file
upload_asset_content
download_asset_file
download_asset_content

Reply:
Done.